### PR TITLE
Fixed issue #15985: print answers: list with comments, comments not printed

### PR DESCRIPTION
**Pull request by gabrieljenik**
This  was implemented on getFullResponseTable(), LSv2.6. 
But seems to got lost in the transitio to LSv3 when moving to SurveyDynamic::getPrintAnswersArray()

Implementing comments for question type 'O'
Rename getComment parameter.

### CODE REVIEW

File: application/models/SurveyDynamic.php [hunk 1]
```diff
@@ -964,13 +964,19 @@ public function getQuestionArray($oQuestion, $oResponses, $bHonorConditions, $su
...
+
         if ($oQuestion->type=='N' || ($oQuestion->parent_qid != 0 && $oQuestion->parents['type'] === "K")) {
             if (strpos($aQuestionAttributes['answervalue'], ".") !== false) { // Remove last 0 and last . ALWAYS (see \SurveyObj\getShortAnswer)
                 $aQuestionAttributes['answervalue'] = rtrim(rtrim($aQuestionAttributes['answervalue'], "0"), ".");
             }
         }
 
+        // If trying to retrieve main question ($getCommentOnly = false), retrieve comment in a new attribute
+        // Check if $getCommentOnly = false to avoid endless recursivity
+        if ($oQuestion->type=='O'&&!$getCommentOnly) {
```
Comment (olleharstedt) [inline application/models/SurveyDynamic.php:976]: PSR-12 recommends (or implies?) space between `==`. It should be:

```php
if ($oQuestion->type == 'O' && !$getCommentOnly) {
```

Please adjust.

File: application/models/SurveyDynamic.php [hunk 2]
```diff
@@ -849,7 +849,7 @@ public function getQuestionArray($oQuestion, $oResponses, $bHonorConditions, $su
         }
 
 
-        if ($getComment === true) {
+        if ($getCommentOnly === true) {
```
Comment (olleharstedt) [inline application/models/SurveyDynamic.php:852]: `=== true` is not needed, should be

```php
if ($getCommentOnly) {
```

File: application/models/SurveyDynamic.php [hunk 3]
```diff
@@ -770,10 +770,10 @@ protected function filterColumns(CDbCriteria $criteria)
      * @param SurveyDynamic $oResponses
      * @param boolean $bHonorConditions
      * @param boolean $subquestion
-     * @param boolean $getComment
+     * @param boolean $getCommentOnly If should only returns the "comments" or "other" response.
```
Comment (olleharstedt) [inline application/models/SurveyDynamic.php:773]: +1 for proper explanation of the argument :)

File: application/models/SurveyDynamic.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/SurveyDynamic.php]: Done

File: application/models/SurveyDynamic.php (hunk 2 above)
Comment (gabrieljenik) [reply application/models/SurveyDynamic.php]: Fixed

**Pull request merged**